zeppelin-interpreter/src/main/java/org/apache/zeppelin/display/Input.java

```diff
@@ -86,6 +86,34 @@ public String getArgument() {
     return argument;
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    Input<?> input = (Input<?>) o;
+
+    if (hidden != input.hidden) return false;
+    if (name != null ? !name.equals(input.name) : input.name != null) return false;
+    if (displayName != null ? !displayName.equals(input.displayName) : input.displayName != null)
+      return false;
+    if (defaultValue != null ?
+        !defaultValue.equals(input.defaultValue) : input.defaultValue != null)
+      return false;
+    return argument != null ? argument.equals(input.argument) : input.argument == null;
+
+  }
+
+  @Override
+  public int hashCode() {
+    int result = name != null ? name.hashCode() : 0;
+    result = 31 * result + (displayName != null ? displayName.hashCode() : 0);
+    result = 31 * result + (defaultValue != null ? defaultValue.hashCode() : 0);
+    result = 31 * result + (hidden ? 1 : 0);
+    result = 31 * result + (argument != null ? argument.hashCode() : 0);
+    return result;
+  }
+
```

Comment:
For testing purpose, so that we can compare GUI after serialization/deserialization.